tests/tracer/test_ext.py

```diff
@@ -23,5 +26,118 @@ def _updateenv(monkeypatch, env):
 
 @pytest.mark.parametrize("name,environment,tags", _ci_fixtures())
 def test_ci_providers(monkeypatch, name, environment, tags):
+    """Make sure all provided environment variables from each CI provider are tagged correctly."""
     _updateenv(monkeypatch, environment)
-    assert tags == ci.tags(environment), "wrong tags in {0} for {1}".format(name, environment)
+    extracted_tags = ci.tags(environment)
+    for key, value in tags.items():
+        assert extracted_tags[key] == value, "wrong tags in {0} for {1}".format(name, environment)
+
+
+def test_git_extract_user_info_author():
```

Comment:
how are other teams testing these?

I wonder if we should do a temp directory with `git init`, `git config ...`, and `git commit ...` if that would be a better integration test?

Reply:
Looks like the [js](https://sourcegraph.com/github.com/DataDog/dd-trace-js/-/blob/packages/dd-trace/test/plugins/util/git.spec.js#L3:7) and [java](https://sourcegraph.com/github.com/DataDog/dd-trace-java/-/blob/internal-api/src/test/groovy/datadog/trace/bootstrap/instrumentation/ci/git/LocalFSGitInfoExtractorTest.groovy#L18:5) team did something similar to us (i.e. mock subprocess calls).

Reply:
hmm, I'd personally prefer an integration test, but this is ok for now.